test-data/unit/check-statements.test

```diff
@@ -1663,3 +1663,19 @@ N = TypedDict('N', {'x': int})
 [builtins fixtures/dict.pyi]
 [typing fixtures/typing-full.pyi]
 [out]
+
+[case testGlobalWithoutInitialization]
+# flags: --new-semantic-analyzer
+from typing import List
+
+def foo() -> None:
+    global bar
+    # TODO: Confusing error message
+    bar = []  # type: List[str]  # E: Name 'bar' already defined (possibly by an import)
```

Comment:
I think this deserves an issue. This is kind of a "fundamental" limitation of the new semantic analyzer. So we should give a dedicated error message and explicitly document that users must write something like this:
```python
bar: List[str]
def foo() -> None:
    global bar
    bar = []
```
Also note that the Python 3.6 version of this test case is a syntax error:
```python
from typing import List

 def foo() -> None:
    global bar
    bar: List[str] = []  # SyntaxError: annotated name 'bar' can't be global
```

Reply:
Good points, I'll create a follow-up issue.

Reply:
Here's the issue: #6976